src/main/scala/beam/router/graphhopper/GraphHopperWrapper.scala

```diff
@@ -62,93 +64,112 @@ class GraphHopperWrapper(
     val alternatives = if (response.hasErrors) {
       Seq()
     } else {
-      response.getAll.asScala.map(responsePath => {
-        var linkIds = IndexedSeq.empty[Int]
-        var totalTravelTime = (responsePath.getTime / 1000).toInt
-        val ghLinkIds: IndexedSeq[Int] =
-          responsePath.getPathDetails
-            .asScala(Parameters.Details.EDGE_ID)
+      response.getAll.asScala
+        .map(responsePath => {
+          var linkIds = IndexedSeq.empty[Int]
+          var totalTravelTime = (responsePath.getTime / 1000).toInt
+          val ghLinkIds: IndexedSeq[Int] =
+            responsePath.getPathDetails
+              .asScala(Parameters.Details.EDGE_ID)
+              .asScala
+              .map(pd => pd.getValue.asInstanceOf[Int])
+              .toIndexedSeq
+
+          val allLinkTravelBeamTimes = responsePath.getPathDetails
+            .asScala(BeamTimeDetails.BEAM_TIME)
+            .asScala
+            .map(pd => pd.getValue.asInstanceOf[Long].toDouble / 1000.0)
```

Comment:
nitpick: would it be better to move this logic with casting and unit conversion to `AbstractBeamTimeDetails` because it is the one who writes the data?